### PR TITLE
First example, CI and README

### PR DESCRIPTION
**Pull request by kjvalencik**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,40 @@
...
+| ---------------------------- | ------------------------------------------ |
+| [`hello-world`][hello-world] | Return a JS String with a greeting message |
+
+[hello-world]: examples/hello-world
+
+## Contributing
+
+### Setup
+
+[`pnpm`](https://www.npmjs.com/package/pnpm) provides monorepo workspace tooling.
```
Comment (dherman) [inline README.md]: Is it worth mentioning that Neon doesn't depend on pnpm, it's just what we're using for this repo?

**Review by dherman — APPROVED** [review]
Looks good!

**Comment by amilajack** [comment]
This project already has a readme and ci. Can you clarify the PR title

File: README.md (hunk 1 above)
Comment (amilajack) [reply README.md]: NPM now has native workspaces support:
https://docs.npmjs.com/cli/v7/using-npm/workspaces

**Comment by kjvalencik** [comment]
@amilajack I moved the current examples to the `legacy` branch (still the default branch) and created a new `main` branch with *no* git history. The intent is for this new branch to only contain `N-API` examples and change it to the default when ready.

This PR is against the empty `main` branch.

**Comment by amilajack** [comment]
Moving all the docs to the legacy branch will break a number of docs which reference the previous examples. Are there plans to port back the previous examples?

File: README.md (hunk 1 above)
Comment (kjvalencik) [reply README.md]: Unfortunately, it doesn't provide the functionality we need. The workspace features built into `npm` are very minimal. They provide linking and that's about it.

We at least need recursive building, but also benefit from module deduplication.

**Comment by kjvalencik** [comment]
@amilajack Good catch! I can update the links so that they only use relative links. That should fix internal links.

Do you mean docs on neon-bindings.com?

File: README.md (hunk 1 above)
Comment (amilajack) [reply README.md]: I see. Which examples require being built recursively?

**Comment by amilajack** [comment]
Docs as in our guide docs, neon-bindings.com

**Comment by kjvalencik** [comment]
@amilajack Is that something you can help me update? I'm working on guiding users towards N-API now that it's getting closer to ready.

**Comment by amilajack** [comment]
Yea I can migrate the guide docs.

**Comment by kjvalencik** [comment]
I fixed README links in this commit. https://github.com/neon-bindings/examples/commit/e7b5e3c4ed35f6c5d84895c2f7e7b06389989b6f

File: README.md (hunk 1 above)
Comment (kjvalencik) [reply README.md]: Not really recursively, but the `examples/*` glob. The `npm` workspaces feature won't run `npm install` on all of them.

I really hope I'm wrong. 😁  I would love to remove `pnpm` and make the examples repo a little easier to understand, if possible.

Comment (amilajack) [reply README.md]: Can work around that by following the existing structure and using explicit paths?

```json
  "workspaces": [
    "a",
    "b"
  ]
```

Comment (kjvalencik) [reply README.md]: I just learned that you can use globs with workspace commands! `npm test --workspace=*` will run on all packages. That might be better than introducing `pnpm`.

**Pull request merged**